cookbooks/aws-parallelcluster-platform/attributes/platform.rb

```diff
@@ -13,6 +13,9 @@
 default['cluster']['nvidia']['enabled'] = 'no'
 default['cluster']['nvidia']['driver_version'] = '470.182.03'
 
+# Spack
+default['cluster']['spack_root'] = "#{node['cluster']['base_dir']}/spack"
```

Comment:
This should be on a shared storage location intended for the user.  I am thinking in the default user's home or something similar.  

['cluster']['base_dir'] is going to be in /opt which is intended to be an internal use location for pcluster